Userland/Games/Minesweeper/main.cpp

```diff
@@ -119,18 +120,36 @@ int main(int argc, char** argv)
     }));
 
     auto& difficulty_menu = window->add_menu("&Difficulty");
-    difficulty_menu.add_action(GUI::Action::create("&Beginner", { Mod_Ctrl, Key_B }, [&](auto&) {
+    GUI::ActionGroup difficulty_actions;
+    difficulty_actions.set_exclusive(true);
+
+    auto action = GUI::Action::create_checkable("&Beginner", { Mod_Ctrl, Key_B }, [&](auto&) {
         field.set_field_size(9, 9, 10);
-    }));
-    difficulty_menu.add_action(GUI::Action::create("&Intermediate", { Mod_Ctrl, Key_I }, [&](auto&) {
+    });
+    action->set_checked(field.rows() == 9 && field.columns() == 9 && field.mine_count() == 10);
+    difficulty_menu.add_action(action);
+    difficulty_actions.add_action(action);
+
+    action = GUI::Action::create_checkable("&Intermediate", { Mod_Ctrl, Key_I }, [&](auto&) {
         field.set_field_size(16, 16, 40);
-    }));
-    difficulty_menu.add_action(GUI::Action::create("&Expert", { Mod_Ctrl, Key_E }, [&](auto&) {
+    });
+    action->set_checked(field.rows() == 16 && field.columns() == 16 && field.mine_count() == 40);
+    difficulty_menu.add_action(action);
+    difficulty_actions.add_action(action);
+
+    action = GUI::Action::create_checkable("&Expert", { Mod_Ctrl, Key_E }, [&](auto&) {
         field.set_field_size(16, 30, 99);
-    }));
-    difficulty_menu.add_action(GUI::Action::create("&Madwoman", { Mod_Ctrl, Key_M }, [&](auto&) {
+    });
+    action->set_checked(field.rows() == 16 && field.columns() == 30 && field.mine_count() == 99);
+    difficulty_menu.add_action(action);
+    difficulty_actions.add_action(action);
+
+    action = GUI::Action::create_checkable("&Madwoman", { Mod_Ctrl, Key_M }, [&](auto&) {
         field.set_field_size(32, 60, 350);
-    }));
+    });
+    action->set_checked(field.rows() == 32 && field.columns() == 60 && field.mine_count() == 350);
```

Comment:
Instead of repeating all of these hard-coded values, it would be way better to add `Field::set_difficulty` setter and `Field::difficulty` getter APIs that take/return some `Difficulty` enumeration.